dev/SampleWinRT/SampleWinRT.h

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include <Common.g.h>
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    struct Common
+    {
+        Common() = default;
+
+        static void Initialize()
+        {
+            std::call_once(m_initOnce, InitInner);
+        }
+
+        static hstring AppIdentity()
+        {
+            Initialize();
+            return { m_fullName.data(), m_fullNameLen };
+        }
+
+        static bool IsAppContainer()
+        {
+            static bool isAppContainer = wil::get_token_is_app_container();
+            return isAppContainer;
+        }
+
+        static bool HasIdentity()
+        {
+            Initialize();
+            return m_identityAvailable;
+        }
+
+        static void InitInner()
+        {
+            // Initially, our only option is to get the package identity from this process
+            if (::GetCurrentPackageFullName(&m_fullNameLen, m_fullName.data()) == ERROR_SUCCESS)
```

Comment:
If you're going to cache something, cache the resulting hstring and keep the `m_fullName`/`m_fullNameLen` local to this initialization function. You can probably also replace `m_identityAvailable` with a check for whether the hstring is empty. It's just easier to maintain when the static surface area of an implementation is as small as possible.